lldb/tools/lldb-dap/ProgressEvent.cpp

```diff
@@ -77,16 +77,19 @@ ProgressEvent::Create(uint64_t progress_id, std::optional<StringRef> message,
   if (event.GetEventType() == progressStart && event.GetEventName().empty())
     return std::nullopt;
 
-  if (prev_event && prev_event->EqualsForIDE(event))
+  if (prev_event && prev_event->EqualsForIDE(event, total))
     return std::nullopt;
 
   return event;
 }
 
-bool ProgressEvent::EqualsForIDE(const ProgressEvent &other) const {
+bool ProgressEvent::EqualsForIDE(const ProgressEvent &other, uint64_t total) const {
   return m_progress_id == other.m_progress_id &&
-         m_event_type == other.m_event_type &&
-         m_percentage == other.m_percentage;
+         m_event_type == other.m_event_type && 
+         // If we check the percentage of a non-deterministic event
```

Comment:
@JDevlieghere & @ashgti, thoughts on also making it 500 ms between non-deterministic progress events, or do we think 250 is the best tradeoff of performance to responsiveness?

Reply:
Isn't `m_percentage` a nullopt when the `total` is UINT64_MAX? Or is there a logic issue in lines 35-65?

Reply:
You're correct, but then we'd be comparing if `null_opt == null_opt`, and no further updates would be sent. I overlooked this when adding the total, we should short circuit and ignore the percentage if percentage is a null opt.

Reply:
But won't they have different event types if its changed? For a non-determinitic progress event, it should only go from `m_event_type == progressStart` to `m_event_type == progressEnd`, right? If we have 2 starts or 2 ends that feels like the logic got mixed up somewhere else, right?

Reply:
First, let me apologize for being overly brief in my explanation. You're correct @ashgti, but now imagine all the updates.

Let's say we're indexing dwarf, you'll get the start, *the first update* and then the end.

As an extreme hypothetical, if this takes 30 minutes you will have 0 progress actually sent to the progress bar in VSCode resulting in what appears to be a hang of LLDB.

So, for non-deterministic events we want to update for every time window. DAP limits to 250 ms, which I think is fine. I might need to change the equals check now that I spell this out but I hope this explains the intent!

Reply:
>  should we add m_message to the Equals check?

Correct, I was using my progress tester which I realized every event has it's own unique message. But it is optional. Let me fix that...